tests/unittests/CoreGraphics/CGFontTests.mm

```diff
@@ -106,4 +106,38 @@
     CGFontRef font = CGFontCreateWithFontName(c_arialBoldItalicName);
     CFAutorelease(font);
     EXPECT_EQ(-434, CGFontGetDescent(font));
+}
+
+TEST(CGFont, CreateWithDataProvider) {
+    // get test startup full path
+    wchar_t fullPath[_MAX_PATH];
+    size_t len = GetModuleFileNameW(NULL, fullPath, _MAX_PATH);
+
+// OSX has twice-as-wide wchars
+#if TARGET_OS_WIN32
+    NSString* testFileFullPath = [NSString stringWithCharacters:(const unichar*)fullPath length:len + 1];
+#else
+    NSString* testFileFullPath = [NSString stringWithBytes:fullPath length:sizeof(wchar_t) * len encoding:WCHAR_ENCODING];
+#endif
+
+    // reconstruct fullpath for test artifact file. e.g., /Volumes/WinObjC/build/Tests/UnitTests/Foundation/OSX/data/NSFileManagerUT.txt
+    testFileFullPath = [[testFileFullPath stringByDeletingLastPathComponent] stringByAppendingString:@"/data/WinObjC-Regular.ttf"];
```

Comment:
If that does not satisfy the usage, make this into a utility function (given it looks like a general sequence) so it can be reused in the future.

---
In reply to: [88152656](https://github.com/Microsoft/WinObjC/pull/1386#discussion_r88152656) [](ancestors = 88152656)